### ISSUE TITLE
Show White Space and TAB does not show all whitespace

### ISSUE THREAD

**Issue by EffectivelyEfficient** [notepad-plus-plus/notepad-plus-plus#827]
"View | Show Symbol | Show White Space and TAB" does not show all white space. This also affects "Show All Characters"

To reproduce, create an html page that contains &#8203;&_#_8203;&#8203; (without the underscores) and then copy the character to notepad++. [If this site allows it, the offending char might also be in this paragraph]

8203 is a Unicode character also know as ZERO WIDTH SPACE (ZWSP). This character is intended for invisible word separation and for line break control; it has no width, but its presence between two characters does not prevent increased letter spacing in justification.

**Comment by dg9ngf** [comment]
There are more whitespace characters in Unicode. Perhaps the most interesting are:

* U+0020 "Space"
* U+00A0 "No-Break Space"
* U+202F "Narrow No-Break Space"

The normal space is the only of them that is visualised. But it's hardly visible so tiny is the symbol for it. It should definitely be a thicker graphic.

The other two mentioned should have different symbols. Microsoft Word for example shows something like ° for the non-breaking space, but also nothing for the narrow non-breaking space.

There's a more complete overview in [Wikipedia](https://de.wikipedia.org/wiki/Leerzeichen#Desktop-Publishing_und_Webtypographie) (not available in English).

**Comment by aaaxx** [comment]
It would also be a good idea to have those other spaces as a separate item in Style Configurator, so that you can have normal spaces dimmed and special ones popping out.

**Comment by Daksol** [comment]
See also feature request #4731, also asking for enhanced visuslisation. This feature request #827 is better as it covers a wider range of problematic characters.

**Comment by MetaChuh** [comment]
we are closing this issue due to its age and multiple newer notepad++ versions since it was filed.

please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html).
if your issue persists, and it is still of relevance to you, please feel free to file a new issue.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
```
[<kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd>](https://notepad-plus-plus.org/community/)

**Comment by ygoe** [comment]
@MetaChuh Still not fixed, still a problem, please reopen.

**Comment by Daksol** [comment]
I also raised this issue via item #4731, focusing on how NPP deals with the zero width character #8203 which fails to show even in View All Chars mode.
But not seen any response since.Thanks.
Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus
 
Description of the Issue Character Decimal 8203, Hex 200B can be put into a text file when screen scraping from certain webpages. These chars are invisible een when in View->Show All Chars mode....
  |   |

  |

  |

  

Sent from Yahoo Mail on Android 
 
  On Wed, 10 Apr 2019 at 1:33, MetaChuh<notifications@github.com> wrote:   
Closed #827.

**Comment by sasumner** [comment]
How do Scintilla versions newer than Notepad++ 7.6.6's "Scintilla 3.5.6" handle this?  If they handle it no differently than this issue describes there isn't much N++ can do and an issue should be raised against Scintilla.  However, if newer Scintilla handles these cases, these specific points should be added as comments to a Notepad++ "upgrade Scintilla" issue.  After this is investigated, closure (oh, good, I see that has already been done!) is the right action for this issue.

**Comment by Daksol** [comment]
I also raised this issue via item #4731, focusing on how NPP deals with the zero width character #8203 which fails to show even in View All Chars mode.
Could you add your note about Scintilla to that item? Else maybe close #4731 as well, but open a new one as you suggested.
(I am keen regular user, but limited understanding of NPP's technical foundations).
Sent from Yahoo Mail on Android 
 
  On Thu, 11 Apr 2019 at 13:42, John Edge<jr_edge@yahoo.com> wrote:   I also raised this issue via item #4731, focusing on how NPP deals with the zero width character #8203 which fails to show even in View All Chars mode.
But not seen any response since.Thanks.
Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus
 
Description of the Issue Character Decimal 8203, Hex 200B can be put into a text file when screen scraping from certain webpages. These chars are invisible een when in View->Show All Chars mode....
  |   |

  |

  |

  

Sent from Yahoo Mail on Android 
 
  On Wed, 10 Apr 2019 at 1:33, MetaChuh<notifications@github.com> wrote:   
Closed #827.

**Comment by double-standard-you** [comment]
> I also raised this issue via item #4731, focusing on how NPP deals with the zero width character #8203 which fails to show even in View All Chars mode. Could you add your note about Scintilla to that item? Else maybe close #4731 as well, but open a new one as you suggested. (I am keen regular user, but limited understanding of NPP's technical foundations). Sent from Yahoo Mail on Android On Thu, 11 Apr 2019 at 13:42, John Edge<jr_edge@yahoo.com> wrote: I also raised this issue via item #4731, focusing on how NPP deals with the zero width character #8203 which fails to show even in View All Chars mode. But not seen any response since.Thanks. Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus | | | | | | | | | | | Feature Request - Visualize Zero-width spaces in "show all chars" mode. · Issue #4731 · notepad-plus-plus/notepad-plus-plus Description of the Issue Character Decimal 8203, Hex 200B can be put into a text file when screen scraping from certain webpages. These chars are invisible een when in View->Show All Chars mode.... | | | | Sent from Yahoo Mail on Android On Wed, 10 Apr 2019 at 1:33, MetaChuh<notifications@github.com> wrote: Closed #827. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

American Shares Truth About Joshua Wong & Hong Kong Democracy
https://www.youtube.com/watch?v=8nXz66Btfl0

American Shares Truth on Xinjiang and Tibet Controversies in China
https://www.youtube.com/watch?v=CldtoYwPtMk

Why is Ukraine the West’s Fault? Featuring John Mearsheimer
https://www.youtube.com/watch?v=JrMiSQAGOS4

Bat coronavirus found in U.S.-funded bio-lab in Ukraine: Russian Defense Ministry
https://www.youtube.com/watch?v=BCydUeHAhzQ

Ukrainians were denied US visas as White House promise help to refugees
https://www.youtube.com/watch?v=BGZhzNFmyhc

Reports of racist treatment against Africans trying to flee Ukraine
https://www.youtube.com/watch?v=JItYOW_uT8s

Fleeing African and Indian students face racism at Ukraine border
https://www.youtube.com/watch?v=ODMOzwI__zs

Canadian RCMP use horses to trample protestors
https://www.youtube.com/watch?v=Usn3CR0E1v0

A bald-faced lie from the Western media corp: Critical “Uyghur Genocide” Questions: A CBC Interview Follow-up
https://www.youtube.com/watch?v=O7IbsPFM2Xg
